src/dft/backends/descriptor.cpp

```diff
@@ -28,7 +28,10 @@ namespace dft {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::commit(sycl::queue &queue) {
-    pimpl_.reset(detail::create_commit(*this, queue));
+    if (!pimpl_) {
+        pimpl_.reset(detail::create_commit(*this, queue));
+    }
+    pimpl_->commit(values_);
```

Comment:
We need to check that the same queue is being used here as for a previous call.

Reply:
I had a think about it and decided that if the queue is changed at all the the whole plan can be reconstructed. If there are performance problem with this (opportunities where the plan could have been shared) we can deal with that later.